framework/packages/abstract-interface/src/deployers.rs

```diff
@@ -17,6 +19,39 @@ pub trait RegisteredModule {
     fn module_version<'a>() -> &'a str;
 }
 
+/// Trait to access module dependency information tied directly to the type.
+pub trait DependencyCreation {
+    /// Type that exposes the dependencies's configurations if that's required.
+    type DependenciesConfig;
+
+    /// Function that returns the `ModuleInstallConfig` for each dependent module.
+    #[allow(unused_variables)]
+    fn dependency_install_configs(
+        configuration: Self::DependenciesConfig,
+    ) -> Result<Vec<ModuleInstallConfig>, crate::AbstractInterfaceError> {
+        Ok(vec![])
+    }
+}
+
+/// Trait to make it easier to construct `ModuleInfo` and `ModuleInstallConfig` for a
+/// `RegisteredModule`.
+pub trait InstallConfig: RegisteredModule {
+    /// Constructs the `ModuleInfo` by using information from `RegisteredModule`.
+    fn module_info() -> Result<ModuleInfo, crate::AbstractInterfaceError> {
+        ModuleInfo::from_id(Self::module_id(), Self::module_version().into()).map_err(Into::into)
+    }
+
+    /// Constructs the ModuleInstallConfig for an App Interface
```

Comment:
Rust is awesome and it support linking to types directly, and the links actually work in an IDE or when reading the docs on docs.rs


```suggestion
    /// Constructs the [`ModuleInstallConfig`] for an App Interface
```

So whenever referring to types in doc-comments, try to link to the type! 
https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html

It also supports MD and code-blocks and other cool things.

Reply:
That is awesome, thanks for the tip!